packages/flutter/test/widgets/editable_text_test.dart

```diff
@@ -12544,6 +12544,176 @@ void main() {
+        expect(state.spellCheckEnabled, isTrue);
+        expect(
+          state.spellCheckConfiguration!.spellCheckService.runtimeType,
+          equals(DefaultSpellCheckService),
+        );
+        expect(
+          state.spellCheckConfiguration!.spellCheckSuggestionsHandler.runtimeType,
+          equals(DefaultSpellCheckSuggestionsHandler),
+        );
+        tester.binding.platformDispatcher.clearNativeSpellCheckServiceDefined();
+    });
+
+    testWidgets(
+      'Spell check configured properly with specified spell check service and handler',
+        (WidgetTester tester) async {
+      final FakeSpellCheckService fakeSpellCheckService = FakeSpellCheckService();
+      final FakeSpellCheckSuggestionsHandler fakeSpellCheckSuggestionsHandler =
+      FakeSpellCheckSuggestionsHandler();
+
+      await tester.pumpWidget(
+        MaterialApp(
+          home: EditableText(
+            controller: TextEditingController(text: 'A'),
+            focusNode: FocusNode(),
+            style: const TextStyle(),
+            cursorColor: Colors.blue,
+            backgroundCursorColor: Colors.grey,
+            cursorOpacityAnimates: true,
+            autofillHints: null,
+            spellCheckConfiguration:
+              SpellCheckConfiguration(
+                spellCheckService: fakeSpellCheckService,
+                spellCheckSuggestionsHandler: fakeSpellCheckSuggestionsHandler,
+                misspelledTextStyle: TextField.materialMisspelledTextStyle,
+            ),
+          ),
+        ),
+      );
+
+      final EditableTextState state =
+        tester.state<EditableTextState>(find.byType(EditableText));
+      expect(
+        state.spellCheckConfiguration!.spellCheckService.runtimeType,
+        equals(FakeSpellCheckService),
+      );
+      expect(
+        state.spellCheckConfiguration!.spellCheckSuggestionsHandler.runtimeType,
+        equals(FakeSpellCheckSuggestionsHandler),
+      );
+    });
+
+    testWidgets(
+      'Error thrown when spell check enabled but no default spell check service available',
+        (WidgetTester tester) async {
+        tester.binding.platformDispatcher.nativeSpellCheckServiceDefinedTestValue =
+          false;
+
+        await tester.pumpWidget(
+            EditableText(
+                controller: TextEditingController(text: 'A'),
+                focusNode: FocusNode(),
+                style: const TextStyle(),
+                cursorColor: Colors.blue,
+                backgroundCursorColor: Colors.grey,
+                cursorOpacityAnimates: true,
+                autofillHints: null,
+                spellCheckConfiguration:
+                  SpellCheckConfiguration(
+                    misspelledTextStyle: TextField.materialMisspelledTextStyle,
+                ),
+            ));
+
+        expect(tester.takeException(), isA<AssertionError>());
+        tester.binding.platformDispatcher.clearNativeSpellCheckServiceDefined();
```

Comment:
Doesn't this test conflict with the test above "Spell check configured properly when spell check enabled without specified spell check service and handler"?  It seems like they're testing the same thing, this one saying it should error and that one saying it shouldn't.  Maybe I'm misreading it though.

Reply:
The difference is that I set `nativeSpellCheckServiceDefined` to be false here and true in the other test. I'll update the titles to make it clearer!

Reply:
Ah thanks, I missed that!